main.tf

```diff
@@ -100,69 +100,69 @@ resource "aws_instance" "int_tableau_linux" {
 
+#Pull values from Parameter Store and save to profile
+sudo touch /home/tableau_srv/env_vars.sh
+echo "
+export DATA_ARCHIVE_TAB_INT_BACKUP_URL=`aws --region eu-west-2 ssm get-parameter --name data_archive_tab_int_backup_url --query 'Parameter.Value' --output text`
+export TAB_INT_REPO_URL=`aws --region eu-west-2 ssm get-parameter --name tab_int_repo_url --query 'Parameter.Value' --output text`
+export TAB_SRV_USER=`aws --region eu-west-2 ssm get-parameter --name tableau_server_username --query 'Parameter.Value' --output text`
+export TAB_SRV_PASSWORD=`aws --region eu-west-2 ssm get-parameter --name tableau_server_password --query 'Parameter.Value' --output text`
+export TAB_ADMIN_USER=`aws --region eu-west-2 ssm get-parameter --name tableau_admin_username --query 'Parameter.Value' --output text`
+export TAB_ADMIN_PASSWORD=`aws --region eu-west-2 ssm get-parameter --name tableau_admin_password --query 'Parameter.Value' --output text`
+" > /home/tableau_srv/env_vars.sh
+
+#Download SSH Key pair to allow DevOps Engineers to log in to the server
+aws --region eu-west-2 ssm get-parameter --name tableau_linux_ssh_private_key --query 'Parameter.Value' --output text --with-decryption > /home/tableau_srv/.ssh/id_rsa
+aws --region eu-west-2 ssm get-parameter --name tableau_linux_ssh_public_key --query 'Parameter.Value' --output text --with-decryption > /home/tableau_srv/.ssh/id_rsa.pub
+
+#Change ownership and permissions of tableau_srv files
+chown -R tableau_srv:tableau_srv /home/tableau_srv/
+chmod 0400 /home/tableau_srv/.ssh/id_rsa
+chmod 0444 /home/tableau_srv/.ssh/id_rsa.pub
+chmod 0644 /home/tableau_srv/env_vars.sh
+
+echo "
+source /home/tableau_srv/env_vars.sh
+" >> /home/tableau_srv/.bashrc
+
+su - tableau_srv
+
 #Initialise TSM (finishes off Tableau Server install/config)
 /opt/tableau/tableau_server/packages/scripts.*/initialize-tsm --accepteula -f -a tableau_srv
 
-source /etc/profile.d/tableau_server.sh
-tsm register --file /tmp/install/tab_reg_file.json
-
+#Run all TSM commands as tableau_srv
+su - tableau_srv
 
-aws --region eu-west-2 ssm get-parameter --name tableau_linux_ssh_private_key --query 'Parameter.Value' --output text --with-decryption > /home/tableau_srv/.ssh/gitlab_key
-chmod 0400 /home/tableau_srv/.ssh/gitlab_key
+tsm licenses activate --trial
+#tsm licenses activate --license-key <PRODUCT_KEY>
 
-su - tableau_srv
-aws --region eu-west-2 ssm get-parameter --name tableau_linux_ssh_public_key --query 'Parameter.Value' --output text --with-decryption > /home/tableau_srv/.ssh/gitlab_key.pub
-chmod 0444 /home/tableau_srv/.ssh/gitlab_key.pub
+tsm register --file /tmp/install/tab_reg_file.json -p $TAB_SRV_PASSWORD
+tsm settings import -f /opt/tableau/tableau_server/packages/scripts.*/config.json
+tsm pending-changes apply
+tsm initialize --start-server --request-timeout 1800
+tabcmd --accepteula
+tabcmd initialuser --server 'localhost:80' --username '$TAB_ADMIN_USER' --password '$TAB_ADMIN_PASSWORD'
```

Comment:
`'$TAB_ADMIN_USER'` and `'$TAB_ADMIN_PASSWORD'` will be taken as literal strings.
Suggest to remove `''` wrapper.